src/PhpParser/Node/Value/ValueResolver.php

```diff
@@ -215,7 +215,7 @@ private function getConstExprEvaluator(): ConstExprEvaluator
     }
 
     /**
-     * @return mixed[]
+     * @phpstan-return mixed[]|null
```

Comment:
changing `@return` to `@phpstan-return` should not be a goal of the `ReturnAnnotationIncorrectNullableRector`, should be reverted or change to `@return mixed[]|null`

Reply:
Same as above, `ReturnAnnotationIncorrectNullableRector` did not do this change, I replaced it manually. Using `@phpstan-return` instead of `@return` makes sure that the `UnionTypesRector` will not misfire and remove the annotation entirely, because it thinks it can since it considers it equivalent to `?array`.